lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -181,27 +186,32 @@ def authorize_replace_to_many_relationship
         )
       end
 
-      def authorize_remove_to_many_relationship
+      def authorize_remove_to_many_relationships
         source_resource = @resource_klass.find_by_key(
           params[:resource_id],
           context: context
         )
         source_record = source_resource._model
 
-        relationship_type = params[:relationship_type].to_sym
-        related_resource = @resource_klass
-          ._relationship(relationship_type)
+        related_resources = @resource_klass
+          ._relationship(params[:relationship_type].to_sym)
           .resource_klass
-          .find_by_key(
-            params[:associated_key],
+          .find_by_keys(
+            params[:associated_keys],
             context: context
           )
-        related_record = related_resource._model unless related_resource.nil?
+
+        # find_by_keys doesn't raise if no records, so raise explicitly here
+        if related_resources.empty?
+          fail JSONAPI::Exceptions::RecordNotFound, params[:associated_keys]
```

Comment:
Do we need this check here? What specs fail if we'd remove this part of the code?

Maybe those tests that would fail have incorrect behaviour which we'd like to change. I don't know, but this seems a bit out of place here.